client/api.go

```diff
@@ -74,23 +74,42 @@ func challengeForPKeyPassword() string {
 	return password
 }
 
-func privateKey(pkeyFilePath string) *rsa.PrivateKey {
+func privateKey(pkeyFilePath string) (*rsa.PrivateKey, error) {
 	keyBytes, err := ioutil.ReadFile(pkeyFilePath)
 	if err != nil {
-		panic(fmt.Sprintf("Unable to load private key from file: %s. Error: %s", pkeyFilePath, err))
+		return nil, fmt.Errorf("Unable to load private key from file: %s. Error: %s \n", pkeyFilePath, err)
 	}
 
-	pKeyPword := challengeForPKeyPassword()
-	key, err := common.PrivateKeyFromBytes(keyBytes, common.String(pKeyPword))
+	var key *rsa.PrivateKey
+	pKeyPword := viper.GetString(config.OraclePassPhrase)
+
+	if !viper.IsSet(config.OraclePassPhrase) {
+		if key = getPrivateKey(keyBytes, pKeyPword, pkeyFilePath); key == nil {
+			pKeyPword = challengeForPKeyPassword()
+		}
+	}
+	key = getPrivateKey(keyBytes, pKeyPword, pkeyFilePath)
+	return key, nil
+}
+
+func getPrivateKey(keyBytes []byte, pKeyPword, pkeyFilePath string) *rsa.PrivateKey {
+	key, err := oci.PrivateKeyFromBytes(keyBytes, oci.String(pKeyPword))
 	if err != nil {
-		panic(fmt.Sprintf("Unable to load private key from file bytes: %s. Error: %s", pkeyFilePath, err))
+		if pKeyPword != "" {
+			fmt.Fprintf(os.Stderr, "Unable to load private key from file bytes: %s. Error: %s \n", pkeyFilePath, err)
+			os.Exit(1)
```

Comment:
`log.Fatal` would be better

Reply:
Propagating the error up would be even better :)